compiler/src/dotty/tools/dotc/ast/tpd.scala

```diff
@@ -573,6 +573,8 @@ object tpd extends Trees.Instance[Type] with TypedTreeInfo {
       Try(tree: Tree)(expr, cases, finalizer)
   }
 
+  override def skipTransform(tree: Tree)(implicit ctx: Context) = tree.tpe.isError
```

Comment:
I was debating what was simpler, and more efficient. I also first thought of overriding `transform`, but: (1) this requires a new `TreeMap` class in `tpd` (2) this adds a stackframe to each node traversal, which can add up. So I think the current approach is more lightweight in the end.